cclient/api/extended/get_account_data.c

```diff
@@ -4,22 +4,43 @@
  *
  * Refer to the LICENSE file for licensing information
  */
-#include "cclient/api/core/find_transactions.h"
-#include "cclient/api/core/get_balances.h"
-#include "common/helpers/sign.h"
 
+#include <inttypes.h>
+#include <utarray.h>
+
+#include "cclient/api/core/get_balances.h"
 #include "cclient/api/extended/get_account_data.h"
+#include "cclient/api/extended/get_new_address.h"
 #include "cclient/api/extended/logger.h"
+#include "common/helpers/sign.h"
+
+static const UT_icd ut_balance_icd = {sizeof(uint64_t), NULL, NULL, NULL};
+
+void account_data_init(account_data_t* const account) {
+  account->balance = 0;
+  memset(account->latest_address, 0, FLEX_TRIT_SIZE_243);
+  account->addresses = NULL;
+  account->transactions = NULL;
+  account->balances = NULL;
+  utarray_new(account->balances, &ut_balance_icd);
+}
+
+void account_data_clear(account_data_t* const account) {
```

Comment:
Why not `account_data_free` ? It doesn't really "clear"

Reply:
it's not exactly freeing the `account_data_t` object.

Reply:
Alright!